src/react-disposable-decorator.test.js

```diff
@@ -65,8 +65,7 @@ describe(`@Cancelable`, () => {
     }
 
     const Comp = Cancelable(mockComponent)
-    const wrapper = shallow(<Comp />)
-    const innerWrapper = wrapper.dive()
+    const wrapper = mount(<Comp />)
```

Comment:
out of curiosity -- does `shallow(<Comp />).dive()` work as an alternative to mount?

Reply:
It should. In Enzyme 3.0+ shallow should run the componentDidMount lifecycles but I had issues getting it to work correctly on the child component's lifecycles. Instead I used Mount which will always run the lifecycles (including the child lifecycles which is what we really care about)